### PR TITLE
[docs] - Migrate Cloud agent guides [CON-340]

### PR DESCRIPTION
**Pull request by erinkcochran87**
This PR:

- Migrates the agent guides for Cloud. These were previously reviewed by the Cloud team (https://github.com/dagster-io/internal/pull/2594) and haven't changed much since that PR.
- Adds the `/dagster-cloud/deployment` category page
- Adds two new MDXComponents (`ReferenceTable`, `ReferenceTableItem`), used to create structured/HTML tables. [View an example here](https://dagster-git-erin-cloud-agent-docs-elementl.vercel.app/dagster-cloud/deployment/agents/amazon-ecs/manually-provisioning-ecs-agent#user_code_launcher-properties). (Note: The code for this in MDX files isn't pretty - I'm open to suggestions. Ideally descriptions could be in Markdown and then parsed over writing HTML.)

To do:

- [x] Address broken links
- [x] Fully implement `GenerateAgentToken` component
- [x] Address parsing in `ReferenceTableItem` component

### CODE REVIEW

File: docs/content/dagster-cloud/deployment/agents/amazon-ecs/creating-ecs-agent-existing-vpc.mdx
Comment (johannkm) [inline docs/content/dagster-cloud/deployment/agents/amazon-ecs/creating-ecs-agent-existing-vpc.mdx:37]: fancy

File: docs/next/components/mdx/includes/dagster-cloud/GenerateAgentToken.mdx
Comment (johannkm) [inline docs/next/components/mdx/includes/dagster-cloud/GenerateAgentToken.mdx:1]: nit, the agent vs the Dagster Cloud agent

File: docs/next/components/mdx/MDXComponents.tsx
Comment (johannkm) [inline docs/next/components/mdx/MDXComponents.tsx:522]: // intentional?

File: docs/next/components/mdx/MDXComponents.tsx
Comment (johannkm) [inline docs/next/components/mdx/MDXComponents.tsx:19]: nit

**Review by johannkm — COMMENTED** [review]
`ReferenceTable` stuff looks fine to me but I defer to @yuhan

File: docs/next/components/mdx/MDXComponents.tsx
Comment (erinkcochran87) [reply docs/next/components/mdx/MDXComponents.tsx:522]: For now! I'm going to address this before merging.

File: docs/next/components/mdx/MDXComponents.tsx
Comment (erinkcochran87) [reply docs/next/components/mdx/MDXComponents.tsx:19]: Will be addressed before merging :)

File: docs/content/dagster-cloud/deployment/agents.mdx
Comment (erinkcochran87) [reply docs/content/dagster-cloud/deployment/agents.mdx:2]: Good catch, yes it should!

**Comment by erinkcochran87** [comment]
@johannkm If this looks good to you, can you approve? We can always clean up the component later

**Review by johannkm — APPROVED** [review]

**Pull request merged**
